cinder/volume/drivers/nexenta/ns5/iscsi.py

```diff
@@ -171,22 +166,37 @@ def delete_volume(self, volume):
 
         :param volume: volume reference
         """
-
-        url = ('storage/pools/%(pool)s/volumeGroups/%(group)s'
-               '/volumes/%(name)s') % {
-            'pool': self.storage_pool,
-            'group': self.volume_group,
-            'name': volume['name']
-        }
-        field = 'originalSnapshot'
-        origin = self.nef.get('{}?fields={}'.format(url, field)).get(field)
+        path = '%2F'.join([
+            self.storage_pool, self.volume_group, volume['name']])
+        url = 'storage/volumes/%s' % path
+        origin = self.nef.get(url).get('originalSnapshot')
         try:
+            url = 'storage/volumes/%s?snapshots=true' % path
             self.nef.delete(url)
         except exception.NexentaException as exc:
-            vol_path = self._get_volume_path(volume)
-            self.destroy_later_or_raise(exc, vol_path)
-            return
-        self.collect_zfs_garbage(origin)
+            if 'Failed to destroy snap' in exc.kwargs['message']['message']:
+                url = 'storage/snapshots?parent=%s' % path
+                snap_list = []
+                snap_map = {}
+                for snap in self.nef.get(url)['data']:
```

Comment:
Since snap_list is not used in other places, this and the next loop can be combined into one.

Reply:
fixed.